pkg/prometheus/promcfg.go

```diff
@@ -4104,6 +4114,24 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	return cfg, nil
 }
 
+func (cg *ConfigGenerator) generateOTLPConfig(_ *assets.StoreBuilder) yaml.MapItem {
+	cfg := yaml.MapSlice{}
+
+	otlpConfig := cg.prom.GetCommonPrometheusFields().OTLP
+
+	if len(otlpConfig.PromoteResourceAttributes) > 0 {
+		cfg = append(cfg, yaml.MapItem{
```

Comment:
I don't think we need a MapSlice here. This would do the trick:
```suggestion
		cfg := yaml.MapItem{
```

Reply:
Thanks.
To be reserved for other fields for `otlpConfig`